pkg/yamlmeta/type.go

```diff
@@ -137,7 +137,7 @@ func (t *MapType) applySchemaDefaults(foundKeys []interface{}, chk TypeCheck, ma
 		chk.Violations = append(chk.Violations, childCheck.Violations...)
 		err := mapNode.AddValue(val)
 		if err != nil {
-			panic("Unreachable")
+			panic(fmt.Sprintf("Internal inconsistency: adding map item: %s", err))
```

Comment:
Improved! 👍🏼 

In general, when I'm faced with reporting particularly rare/unexpected situations, I try to report as much context as I can.  So, I scan around the function that is going to emit this panic and look for values that are essential to the failed operation and make sure they are either included in the Error message, or report it myself.  For example, is `val` in the message of `err`? And what's the state of `mapNode`?

I think this is okay, for now, but would be worth the little extra effort to make sure those surprising moments come more wrapped in a bow.